scripts/deleteElasticRecords.py

```diff
@@ -28,26 +29,51 @@ def main():
 
     dbManager.createSession()
 
-    batchSize = 200
+    batchSize = 1000
+    esWorkUUIDS = deque()
+    psqlWorkUUIDTuples = deque()
     searchES = Search(index=os.environ['ELASTICSEARCH_INDEX'])
 
-    for work in searchES.query('match_all').scan():    
-        if dbManager.session.query(Work) \
-            .group_by(Work.uuid) \
-            .filter(Work.uuid == work.uuid) \
-            .yield_per(batchSize) == None:
-
-            try:
-                ESWork.delete(work.uuid)
-            except NotFoundError or ValueError or ConflictError:
-                if ValueError:
-                    print('Empty value')
-                elif ConflictError:
-                    print('Version number error')
-                else:
-                    print('Work not indexed, skipping')
+    for work in searchES.query('match_all').scan():
+        # Unpack the tuples to a flat array
+        # Convert the array of ElasticSearch UUIDs and the array of postgresql UUIDs to sets()
+        # Take the difference of of the ES set and the postgresql set (https://docs.python.org/3/library/stdtypes.html#frozenset.difference)
+        # what you have left should be a set of UUIDs that are in ES but not in psql
+            # that's what you should delete!
 
+        esWorkUUIDS.append(work.uuid) # Initialize the array outside of the loop here  
+        if len(esWorkUUIDS) >= batchSize:
+            findESOnlyWorks(esWorkUUIDS, psqlWorkUUIDTuples, dbManager, esManager)  
+            psqlWorkUUIDTuples = deque()
+            esWorkUUIDS = deque()
+
+    findESOnlyWorks(esWorkUUIDS, psqlWorkUUIDTuples, dbManager, esManager) #For any remainder ES works
+    
     dbManager.closeConnection()
 
+def findESOnlyWorks(esWorkUUIDS, psqlWorkUUIDTuples, dbManager, esManager):
+        for uuid in esWorkUUIDS:
+            uuidTuple = dbManager.session.query(Work.uuid) \
+                .filter(Work.uuid == uuid) \
+                .filter(Work.uuid.in_(esWorkUUIDS)).first()
+            if uuidTuple != None:
+                psqlWorkUUIDTuples.append(str(uuidTuple[0])) #PSQL work uuids are of type uuid unlike ES work uuids
```

Comment:
I think we should be able to avoid the need for a loop here and just use `extend` with a list comprehension or something similar.

Or main thing is executing only one database query, how the UUIDs make their way into `psqlWorkUUIDTuples` is less concerning from a performance standpoint

Reply:
Are you referring to do something like this: ```psqlWorkUUIDTuples.update({uuid for uuid in esWorkUUIDS if dbManager.session.query(text(uuid)) != None})```